File: matlab_analysisV3/calculateabs.py
def calculateabs(data,time):
    import numpy as np

    # Detect start of peaks.
    inn=np.argwhere(data)
    peakStart=[];
    peakStart[0]=inn[0];
    for ii in range(1,len(inn)):
        if(inn[ii]-inn[ii-1]!=1):
            peakStart.append(inn[ii])

    # Detect end of peaks.
    inn=np.flip(inn,0);
    peakEnd=[];
    peakEnd[0]=inn[0];
    for ii in range(1,len(inn)):
        if(inn[ii-1]-inn[ii]!=1):
            peakEnd.append(inn[ii])

    # Calculate the average maximum for each contraction.
    peakMax=0
    for ii in range(0,len(peakStart)):
        peakMax=peakMax+max(data[peakStart[ii]:peakEnd[ii]])
    peakMax=peakMax/len(peakStart);

    return peakMax;

File: MATLAB/matlab_analysisV3/calculateabs.m
function absresults = calculateabs(data,time)
    figure; plot(time,data);
    
    % Detect start of peaks.
    [in,~]=find(data~=0);
    peakStart=[];
    peakStart(1)=in(1);
    for ii=2:length(in)
        if(in(ii)-in(ii-1)~=1), peakStart(end+1)=in(ii); end
    end
     % Detect end of peaks.
    in=flip(in,1);
    peakEnd=[];
    peakEnd(1)=in(1);
    for ii=2:length(in)
        if(in(ii-1)-in(ii)~=1), peakEnd=cat(2,in(ii),peakEnd); end
    end
    
    % Calculate the average maximum for each contraction.
    peakMax=0;
    for ii=1:length(peakStart)
        peakMax=peakMax+max(data(peakStart(ii):peakEnd(ii)));
    end
    peakMax=peakMax/length(peakStart);









    absresults=peakMax;
end
